File: Frontend/src/services/authService.js
import axiosInstance from "../api/axiosInstance";

const authService = {
  login: async (credentials) => {
    const response = await axiosInstance.post("/auth/login", credentials);
    return response.data;
  },

  register: async (userData) => {
    const response = await axiosInstance.post("/auth/register", userData);
    return response.data;
  },

  forgotPassword: async (email) => {
    const response = await axiosInstance.post("/auth/forgot-password", {
      email,
    });
    return response.data;
  },

  verifyOtp: async (email, otp) => {
    const response = await axiosInstance.post("/auth/verify-otp", {
      email,
      otp,
    });
    return response.data;
  },

  resetPassword: async (
    email,
    newPassword,
    confirmPassword
  ) => {
    const response = await axiosInstance.post("/auth/reset-password", {
      email,
      newPassword,
      confirmPassword,
    });
    return response.data;
  },

    changePassword: async (
    currentPassword,
    newPassword,
    confirmPassword
  ) => {
    const response = await axiosInstance.put("/auth/change-password", {
      currentPassword,
      newPassword,
      confirmPassword,
    });

    return response.data;
  },
};

export default authService;

File: Backend/src/main/java/com/project/autoserve/service/AuthService.java
package com.project.autoserve.service;

import com.project.autoserve.dto.auth.ChangePasswordRequestDTO;
import com.project.autoserve.dto.auth.ForgotPasswordRequestDTO;
import com.project.autoserve.dto.auth.LoginRequestDTO;
import com.project.autoserve.dto.auth.LoginResponseDTO;
import com.project.autoserve.dto.auth.RegisterRequestDTO;
import com.project.autoserve.dto.auth.ResetPasswordRequestDTO;
import com.project.autoserve.dto.auth.VerifyOtpRequestDTO;

public interface AuthService {

    /**
     * Register a new user
     * @param request RegisterRequestDTO containing user details.
     * @return Success message.
     */
    String register(RegisterRequestDTO request);

    /**
     * Authenticate user and generate JWT token.
     * @param request LoginRequestDTO containing email and password.
     * @return LoginResponseDTO containing JWT token and user details.
     */
    LoginResponseDTO login(LoginRequestDTO request);

    void forgotPassword(ForgotPasswordRequestDTO request);

    void verifyOtp(VerifyOtpRequestDTO request);

    String resetPassword(ResetPasswordRequestDTO request);
    
    void changePassword(ChangePasswordRequestDTO request);
}
